server/src/main/java/io/deephaven/server/table/ops/TimeTableGrpcImpl.java

```diff
@@ -48,9 +52,25 @@ public void validateRequest(final TimeTableRequest request) throws StatusRuntime
     public Table create(final TimeTableRequest request,
             final List<SessionState.ExportObject<Table>> sourceTables) {
         Assert.eq(sourceTables.size(), "sourceTables.size()", 0);
-        final long startTime = request.getStartTimeNanos();
-        final long periodValue = request.getPeriodNanos();
+
         return new TimeTable(ExecutionContext.getContext().getUpdateGraph(), scheduler,
-                startTime <= 0 ? null : DateTimeUtils.epochNanosToInstant(startTime), periodValue, false);
+                adaptStartTime(request),
+                adaptPeriod(request),
+                false);
+    }
+
+    private static Instant adaptStartTime(@SuppressWarnings("unused") final TimeTableRequest request) {
+        if (request.hasStartTimeString()) {
+            return DateTimeUtils.parseInstant(request.getStartTimeString());
+        }
+        // Return null if start time nanos is zero or negative.
+        return request.getStartTimeNanos() <= 0 ? null : DateTimeUtils.epochNanosToInstant(request.getStartTimeNanos());
```

Comment:
I suggest using `Long.MIN_VALUE` instead of zero for the null value, since 0 is theoretically in the middle of the valid range of start times.

Reply:
This is a good point, but would break existing clients which expect 0 to mean "no start time"; which I think is the most common case. I wonder if we should create a new RPC for this? I think this may be somewhat an oversight on our part in originally developing this RPC. Or, we can say "use start_time_string if you need exactly the epoch, or before the epoch"... cc @niloc132 .

Reply:
Agreed to restore original behavior and not break existing clients.